storage/storage_image.go

```diff
@@ -0,0 +1,554 @@
+		copy(newblob, sigblob)
+		copy(newblob[len(sigblob):], sig)
+		sigblob = newblob
+	}
+	s.SignatureSizes = sizes
+	defer s.saveMetadata()
+	return s.store.SetImageBigData(s.ID, "signatures", sigblob)
+}
+
+func (s *storageImage) SupportedManifestMIMETypes() []string {
+	return nil
+}
+
+func (s *storageImage) GetBlob(digest string) (rc io.ReadCloser, n int64, err error) {
+	if blob, ok := s.BlobData[digest]; ok {
+		r := bytes.NewReader(blob)
+		return ioutil.NopCloser(r), r.Size(), nil
+	}
+	if _, ok := s.Layers[digest]; !ok {
+		b, err := s.store.GetImageBigData(s.ID, digest)
+		if err != nil {
+			return nil, -1, err
+		}
+		r := bytes.NewReader(b)
+		logrus.Debugf("exporting opaque data as blob %q", digest)
+		return ioutil.NopCloser(r), r.Size(), nil
+	}
+	logrus.Debugf("exporting filesystem layer as blob %q", digest)
+	return s.diffLayer(s.Layers[digest], true)
+}
+
+func (s *storageImage) diffLayer(layerID string, computeSize bool) (rc io.ReadCloser, n int64, err error) {
+	layer, err := s.store.GetLayer(layerID)
+	if err != nil {
+		return nil, -1, err
+	}
+	layerMeta := storageLayerMetadata{
+		ExpectedSize: -1,
+	}
+	if layer.Metadata != "" {
+		if err := json.Unmarshal([]byte(layer.Metadata), &layerMeta); err != nil {
+			logrus.Errorf("error decoding metadata for layer %q: %v", layerID, err)
+			return nil, -1, err
+		}
+	}
+	if computeSize {
+		if layerMeta.ExpectedSize == -1 {
+			n, err = s.store.DiffSize("", layer.ID)
+			if err != nil {
+				return nil, -1, err
+			}
+		} else {
+			n = layerMeta.ExpectedSize
+		}
+	} else {
+		n = -1
+	}
+	diff, err := s.store.Diff("", layer.ID)
+	if err != nil {
+		return nil, -1, err
+	}
+	return diff, n, nil
+}
+
+func (s *storageImage) GetManifest() (manifest []byte, MIMEType string, err error) {
+	manifest, err = s.store.GetImageBigData(s.ID, "manifest")
+	return manifest, "", err
+}
+
+func (s *storageImage) GetTargetManifest(digest string) (manifest []byte, MIMEType string, err error) {
+	requested, err := ddigest.ParseDigest(digest)
+	if err != nil {
+		return nil, "", err
+	}
+	manifest, err = s.store.GetImageBigData(s.ID, "manifest")
+	manifestDigest := requested.Algorithm().FromBytes(manifest)
+	if manifestDigest != requested {
+		return nil, "", ErrManifestDigestMismatch
+	}
+	return manifest, "", err
+}
+
+func (s *storageImage) GetSignatures() (signatures [][]byte, err error) {
+	var offset int
+	if err := s.loadMetadata(); err != nil {
+		logrus.Errorf("error decoding metadata for image: %v", err)
+		return nil, err
+	}
+	signature, err := s.store.GetImageBigData(s.ID, "signatures")
+	if err != nil {
+		return nil, err
+	}
+	sigslice := [][]byte{}
+	for _, length := range s.SignatureSizes {
+		sigslice = append(sigslice, signature[offset:offset+length])
+		offset += length
+	}
+	if offset != len(signature) {
+		return nil, fmt.Errorf("signatures data contained %d extra bytes", len(signatures)-offset)
+	}
+	return sigslice, nil
+}
+
+func (s *storageImage) DeleteImage() error {
```

Comment:
This is no longer necessary here, the interface has bee moved to `storageReference.DeleteImage`.